src/box/lua/net_box.lua

```diff
@@ -138,8 +138,8 @@ local function parse_connect_params(host_or_uri, ...) -- self? host_or_uri port?
                         type(host_or_uri) == 'number' or
                         type(host_or_uri) == 'table') then
         uri = host_or_uri
-    elseif type(host_or_uri) == 'string' and (type(port) == 'string' or
-                                              type(port) == 'number') then
+    elseif (type(host_or_uri) == 'string' or host_or_uri == nil) and
+            (type(port) == 'string' or type(port) == 'number') then
```

Comment:
I don't see a test that would cover this code change. Please add a test for the net.box module.

It'd also be nice to have a test for `console.connect` so that we don't occasionally re-introduce this degradation in future.

Reply:
Added tests for `src/lib/uri` with case `uri.format(uri.parse(3301))`.
Also added tests for `tarantoolctl connect` and `net.box.connect` with only port indication.